benches/benches/state.rs

```diff
@@ -0,0 +1,322 @@
+use criterion::{
+    criterion_group,
+    criterion_main,
+    Criterion,
+};
+use fuel_core::database::{
+    storage::{
+        ContractsStateMerkleData,
+        ContractsStateMerkleMetadata,
+        SparseMerkleMetadata,
+    },
+    vm_database::VmDatabase,
+};
+use fuel_core_storage::{
+    InterpreterStorage,
+    StorageAsMut,
+};
+use fuel_core_types::{
+    fuel_merkle::sparse::MerkleTree,
+    fuel_tx::Bytes32,
+    fuel_types::ContractId,
+};
+use rand::{
+    rngs::StdRng,
+    thread_rng,
+    Rng,
+    SeedableRng,
+};
+use std::iter;
+
+fn setup(db: &mut VmDatabase, contract: &ContractId, n: usize) {
+    let mut rng_keys = thread_rng();
+    let gen_keys = || -> Bytes32 { rng_keys.gen() };
+    let state_keys = iter::repeat_with(gen_keys).take(n);
+
+    let mut rng_values = thread_rng();
+    let gen_values = || -> Bytes32 { rng_values.gen() };
+    let state_values = iter::repeat_with(gen_values).take(n);
+
+    // State key-values
+    let state_key_values = state_keys.zip(state_values);
+
+    // Insert the key-values into the database while building the Merkle tree
+    let tree: MerkleTree<ContractsStateMerkleData, _> =
```

Comment:
Maybe it is worth using `Database::init_contract_state` here, maybe it is be faster=)